ingestion-edge/contract-tests/test_api.py

```diff
@@ -97,7 +101,7 @@ def test_publish(
         rec_time = parse(attrs.pop("submission_timestamp")[:-1])
         assert (req_time - rec_time).total_seconds() < 1
         assert "remote_addr" in attrs
-        attrs.pop("remote_addr")
+        assert attrs.pop("remote_addr") != ""
```

Comment:
I think this would be better as `assert attrs.pop("remote_addr")`, which uses that the empty string is falsey. It would also guard against `None`, if for whatever reason that was passed (though I don't think `None` is likely/possible?)